core/src/main/java/org/pentaho/di/shared/DelegatingSharedObjectsIO.java

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright 2024 Hitachi Vantara.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.pentaho.di.shared;
+
+import org.pentaho.di.core.exception.KettleException;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.w3c.dom.Node;
+
+
+/**
+ * A Read-only wrapper around multiple other SharedObjectsIO implementations that combines them by precedence order.
+ *
+ */
+public class DelegatingSharedObjectsIO implements SharedObjectsIO {
```

Comment:
This looks like a new file. We already have a DelegatingShareObjectsIO

Reply:
ah, the existing one was misspelled... "Share" vs "Shared"... that's why I didn't see it. 

It's unused and doesn't have any tests. 

The two are extremely similar, but the newer one uses `getSharedObject()` instead of `getSharedObjects` for the single-item API, which should be more efficient. I'll delete the old one.

Reply:
It is used in project-plugin. You may have to check that

Reply:
I did check there... I don't see any usage... where is it used specifically? 
I *do* use the new class there.